js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -29,6 +24,10 @@ export interface AuthorizationResultCache {
     set(authorizationResult: AuthorizationResult): Promise<void>;
 }
 
+export interface AddressSelector {
+    select(addresses: AuthorizationResult['addresses']): Promise<AuthorizationResult['addresses'][number]>;
```

Comment:
Maybe a stylistic choice since the resolved type will be the same, but `AuthorizationResult['addresses']`, and definitely `AuthorizationResult['addresses'][number]`, are less clear than their `Base64EncodedAddress[] / Base64EncodedAddress`.

Reply:
Good question. I think there are two positions you can take here:

1. This is a thing that turns `T[]` into `T`
2. This is a thing that takes `ThisParticularType[]` and turns it into `ThisParticularType`

My goal with making it a lookup on `addresses` was to make Typescript do more work for me; if the type changes then that change automatically flows through this function.

My implementation actually hits somewhere in between option 1 and 2, and maybe I should go for option 1 which just says ‘I’m going to give you a list of who knows what and you give me a single item back.’

Reply:
Oh, you like writing Java? Name every `AsyncSelector { select<T>(items: T[]): Promise<T> }`

Reply:
I like not writing types at all and making Typescript work for the money I spent on it.

Uh… wait.